src/modules/peek/Peek.UI/Views/TitleBar.xaml

```diff
@@ -10,16 +10,123 @@
     xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
     mc:Ignorable="d">
 
-    <Grid x:Name="titleBar">
+    <Grid x:Name="TitleBarRootContainer" Height="48">
+        <Grid.ColumnDefinitions>
+            <ColumnDefinition Width="Auto" />
+            <ColumnDefinition Width="*" />
+        </Grid.ColumnDefinitions>
+
         <StackPanel
-            Margin="16,8,8,8"
-            VerticalAlignment="Top"
+            x:Name="AppIconAndName"
+            Grid.Column="0"
+            Margin="8,4"
+            HorizontalAlignment="Left"
+            VerticalAlignment="Center"
             Orientation="Horizontal">
-            <Image Source="../Assets/AppList.png" />
+
+            <Image
+                x:Name="PeekLogo"
+                x:Uid="PeekLogo"
+                Width="24"
+                Height="24"
+                Margin="4,0"
+                VerticalAlignment="Center"
+                Source="../Assets/AppList.png"
+                Stretch="UniformToFill" />
+
             <TextBlock
-                x:Uid="AppTitleText"
-                Margin="12,0,0,0"
-                Style="{StaticResource CaptionTextBlockStyle}" />
+                x:Name="AppTitle_FileCount"
+                x:Uid="AppTitle_FileCount"
+                Margin="4,0,0,0"
+                VerticalAlignment="Center"
+                FontWeight="Bold"
+                Style="{StaticResource CaptionTextBlockStyle}"
+                Text="{x:Bind FileCountText, Mode=OneWay}"
+                Visibility="Collapsed" />
+
+            <TextBlock
+                x:Name="AppTitle_FileName"
+                x:Uid="AppTitle_FileName"
+                MaxWidth="100"
+                Margin="4,0,0,0"
+                VerticalAlignment="Center"
+                Style="{StaticResource CaptionTextBlockStyle}"
+                Text="{x:Bind File.FileName, Mode=OneWay}"
+                TextWrapping="NoWrap" />
         </StackPanel>
+
+        <Grid
+            x:Name="LaunchAppButtonContainer"
+            Grid.Column="1"
+            Margin="4,4,144,4"
+            HorizontalAlignment="Right"
+            VerticalAlignment="Center">
+            <Button
+                x:Name="LaunchAppButton"
+                x:Uid="LaunchAppButton"
+                Click="LaunchAppButton_Click"
```

Comment:
Consider adding a RelayCommand instead (https://learn.microsoft.com/en-us/dotnet/communitytoolkit/mvvm/generators/relaycommand) and add a keyboardaccelerator to launch via "enter".